.github/workflows/rust.yml

```
@@ -52,12 +50,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Install rustup
-        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --profile=minimal
-      - name: Install nightly rust
-        run: |
-          rustup install nightly
-          rustup default nightly
+      - name: Install rust nightly
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: nightly
```

```suggestion
      - uses: dtolnay/rust-toolchain@nightly
```

Same for other places that use a fixed toolchain.

---

Done.